File: project_euler/Problem009/Problem009.java
package project_euler;

public class Problem009 {
    public void solution() {
        int target = 1000;
        for (int i = target / 3; i > 0; --i) {
            for (int j = target >> 1; j > i; --j) {
                int remainder = target - i - j;
                if (i * i + j * j == remainder * remainder) {
                    System.out.println(i * j * remainder);
                    return;
                }
            }
        }
    }
}

File: project_euler/Problem009/Problem009.py
class Problem009:
    def solution(self) -> None:
        target = 1000
        for i in range(target // 3, 0, -1):
            for j in range(target >> 1, i, -1):
                remainder = target - i - j
                if i * i + j * j == remainder * remainder:
                    print(i * j * remainder)
                    return
